zim/notebook/index/pages.py

```diff
@@ -539,6 +539,17 @@ def walk_bottomup(self, path=None):
 		page_id = self._pages.get_page_id(path) if path else ROOT_ID # can raise
 		return self._pages.walk_bottomup(page_id)
 
+	def search_pagename_substring(self, nameFragment):
+		candidates = self.db.execute(
+			'SELECT * FROM pages WHERE LOWER(name) LIKE ? ORDER BY parent,sortkey',
```

Comment:
Original code matches on "basename" - matching on "name" gives different behavior. When you match pages with many children, matching "name" will include all children as well, which is not the intended behavior.

Afraid I don't know immediately the best way to match only the basename in an SQL query. Might need adding another column or using a SQL function to generate the basename.

Probably it would make sense to also add a unittest for this method in order to explicitly check the behavior. See ``tests/indexviews.py`` for similar tests on the ``PagesView`` object.